plaso/parsers/azure_application_gateway_access.py

```diff
@@ -0,0 +1,250 @@
+# -*- coding: utf-8 -*-
+"""Parser for Azure application gateway access logging saved to a file."""
+
+import json
+from json.decoder import JSONDecodeError
+import os
+
+from dfdatetime import semantic_time as dfdatetime_semantic_time
+from dfdatetime import time_elements as dfdatetime_time_elements
+from dfvfs.helpers import text_file
+
+from plaso.containers import events
+from plaso.containers import time_events
+from plaso.lib import errors
+from plaso.lib import definitions
+from plaso.parsers import manager
+from plaso.parsers import interface
+
+
+class AzureApplicationGatewayAccessEventData(events.EventData):
+  """Azure application gateway access log event data.
+
+  Attributes:
+    instance_id (str): Application Gateway instance that served the request
+    client_ip (str): Originating IP for the request.
+    client_port (int): Originating port for the request.
+    http_method (str): HTTP method used by the request.
+    original_request_uri_with_args (str): This field contains the original
+      request URL
+    request_uri (str): URI of the received request.
+    request_query (str):
+      Server-Routed: Back-end pool instance that was sent the request.
```

Comment:
style guide nit: Please start the description after ```(str):``` 
Hanging indents require 4 spaces. Please change throughout this docstring.
(reference: https://github.com/log2timeline/l2tdocs/blob/main/process/Style-guide.md#indentation)